batch7crud-roll2/src/main/java/com/lftechnology/batch7crud/dao/EmployeeDao.java

```diff
@@ -0,0 +1,122 @@
+ */
+public class EmployeeDao {
+  private static final Logger LOGGER = Logger.getLogger(EmployeeDao.class.getName());
+
+  public void insert(Employee employee) throws DataException {
+    try (Connection con = DBConnection.getSqlConnection();
+            PreparedStatement preStmt = con.prepareStatement(QueryConstants.EMPLOYEE_INSERT)) {
+      preStmt.setString(1, employee.getName());
+      preStmt.setString(2, employee.getAddress());
+      preStmt.setString(3, employee.getEmail());
+      preStmt.setString(4, employee.getContact());
+      preStmt.execute();
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public List<Employee> fetch(int noOfRecordsPerPage, int page) throws DataException {
+    try (Connection con = DBConnection.getSqlConnection();
+            PreparedStatement preStmt = con.prepareStatement(QueryConstants.EMPLOYEE_SELECT_LIMIT_OFFSET)) {
+      List<Employee> empList = new ArrayList<Employee>();
+      preStmt.setInt(1, noOfRecordsPerPage);
+      preStmt.setInt(2, page);
+      ResultSet resultSet = preStmt.executeQuery();
+      while (resultSet.next()) {
+        Employee employee = new Employee();
+        employee.setId(resultSet.getInt(EntityConstants.ID));
+        employee.setName(resultSet.getString(EntityConstants.USERNAME));
+        employee.setAddress(resultSet.getString(EntityConstants.ADDRESS));
+        employee.setEmail(resultSet.getString(EntityConstants.EMAIL));
+        employee.setContact(resultSet.getString(EntityConstants.CONTACT));
+        empList.add(employee);
+      }
+      return empList;
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public Employee fetchById(int id) throws DataException {
+    try (Connection con = DBConnection.getSqlConnection();
+            PreparedStatement preStmt = con.prepareStatement(QueryConstants.EMPLOYEE_SELECT_BY_ID)) {
+      Employee employee = null;
+      preStmt.setInt(1, id);
+      ResultSet resultSet = preStmt.executeQuery();
+      while (resultSet.next()) {
+        employee = new Employee();
+        employee.setId(resultSet.getInt(EntityConstants.ID));
+        employee.setName(resultSet.getString(EntityConstants.USERNAME));
+        employee.setAddress(resultSet.getString(EntityConstants.ADDRESS));
+        employee.setEmail(resultSet.getString(EntityConstants.EMAIL));
+        employee.setContact(resultSet.getString(EntityConstants.CONTACT));
+      }
+      return employee;
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public void update(Employee employee) throws DataException {
+    try (Connection con = DBConnection.getSqlConnection();
+            PreparedStatement preStmt = con.prepareStatement(QueryConstants.EMPLOYEE_UPDATE_BY_ID)) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) "" is provided externally to the method and not sanitized before use. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2077)